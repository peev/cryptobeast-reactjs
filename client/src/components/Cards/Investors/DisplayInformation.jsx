// @flow
import React from 'react';
import withStyles from '@material-ui/core/styles/withStyles';

const styles = () => ({
  containerStyle: {
    boxSizing: 'border-box',
    width: '100%',
    border: '1px solid #afb3bb',
    padding: '5px 10px 0 10px',
    marginTop: '20px',
    color: '#afb3bb',
    '& p': {
      margin: '0',
    },
    '& p:first-child': {
      fontSize: '17px',
    },
    '& p:last-child': {
      fontSize: '13px',
      textTransform: 'lowercase',
    },
  },
});

type Props = {
  value: number,
  placeholderText: string,
  classes: Object,
};

const DisplayInformation = ({ ...props }: Props) => {
  const { classes, value, placeholderText } = props;

  return (
    <div className={classes.containerStyle} >
      <p >{value}</p>
      <p >{placeholderText}</p>
    </div>
  );
};

export default withStyles(styles)(DisplayInformation);
