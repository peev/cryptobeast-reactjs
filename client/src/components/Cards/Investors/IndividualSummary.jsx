// @flow
import React from 'react';
import { withStyles, Grid } from 'material-ui';
import { inject, observer } from 'mobx-react';
import { ValidatorForm } from 'react-material-ui-form-validator';
import SelectInvestor from '../../Selectors/SelectInvestor';


const styles = () => ({
  container: {
    width: '100%',
    marginTop: '20px',
    padding: '0 20px',
  },
  containerItems: {
    paddingTop: '20px',
    display: 'flex',
    alignItems: 'baseline',
    justifyContent: 'space-between',
    borderBottom: '1px solid black',
    '&>p': {
      textTransform: 'uppercase',
      fontSize: '14px',
      margin: '0',
      padding: '0',
    },
    '&>span': {
      fontSize: '14px',
      fontWeight: '600',
    },
  },
  overrideGrid: {
    padding: '12px',
  },
  profitStyle: {
    color: '#60bb9b',
  },
});

type Props = {
  classes: Object,
  InvestorStore: Object,
};

@inject('InvestorStore')
@observer
class IndividualSummary extends React.Component<Props> {
  handleSave = () => {
    // console.log('Submit');
  };

  handleSelectInvestorForSummary = (value: *) => {
    this.props.InvestorStore.selectInvestorIndividualSummary(value);
  }

  render() {
    const { classes, InvestorStore } = this.props;

    return (
      <ValidatorForm
        onSubmit={this.handleSave}
      >
        <Grid container>
          <Grid container className={classes.container}>
            <Grid item xs={6} sm={6} md={3}>
              <SelectInvestor
                value={InvestorStore.selectedInvestorIndividualSummaryId || ''}
                handleChange={this.handleSelectInvestorForSummary}
                style={{
                  width: '95%',
                  border: 'none',
                  borderRadius: 0,
                  borderBottom: '1px solid #757575',
                }}
              />
            </Grid>
          </Grid>

          <Grid container className={classes.container}>
            <Grid item xs={12} sm={6} md={3}>
              <div className={classes.containerItems}>
                <p>Shares Held:</p>
                <span>{InvestorStore.individualSharesHeld}</span>
              </div>

              <div className={classes.containerItems}>
                <p>Weighted entry price:</p>
                <span>{InvestorStore.individualWeightedEntryPrice ? `$ ${InvestorStore.individualWeightedEntryPrice}` : ''}</span>
              </div>
            </Grid>

            <Grid item xs={12} sm={6} md={3}>
              <div className={classes.containerItems}>
                <p>USD Equivalent:</p>
                <span>{InvestorStore.individualUSDEquivalent}</span>
              </div>

              <div className={classes.containerItems}>
                <p>BTC Equivalent:</p>
                <span>{InvestorStore.individualBTCEquivalent}</span>
              </div>
            </Grid>

            <Grid item xs={12} sm={6} md={3}>
              <div className={classes.containerItems}>
                <p>Investment Period:</p>
                <span>{InvestorStore.individualInvestmentPeriod ? `${InvestorStore.individualInvestmentPeriod} DAYS` : ''}</span>
              </div>

              <div className={classes.containerItems}>
                <p>Profit:</p>
                <span className={classes.profitStyle}>{InvestorStore.individualProfit ? `${InvestorStore.individualProfit} %` : ''}</span>
              </div>
            </Grid>

            <Grid item xs={12} sm={6} md={3}>
              <div className={classes.containerItems}>
                <p>ETH Equivalent:</p>
                <span>{InvestorStore.individualETHEquivalent}</span>
              </div>

              <div className={classes.containerItems}>
                <p>Fee Potential:</p>
                <span>{InvestorStore.individualFeePotential ? `$ ${InvestorStore.individualFeePotential}` : ''}</span>
              </div>
            </Grid>
          </Grid>
        </Grid>
      </ValidatorForm>
    );
  }
}

export default withStyles(styles)(IndividualSummary);
