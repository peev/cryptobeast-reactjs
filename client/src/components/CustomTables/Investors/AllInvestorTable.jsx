// @flow
import React from 'react';
import {
  withStyles,
  Table,
  TableHead,
  TableRow,
  TableBody,
  TableCell,
  Tooltip,
  TableSortLabel,
} from '@material-ui/core';
import { inject, observer } from 'mobx-react';
import uuid from 'uuid/v4';
import moment from 'moment';
import tableStyle from '../../../variables/styles/tableStyle';
import AssignInvestor from '../../../components/Modal/InvestorModals/AssignInvestor';
import combineStyles from '../../../features/CombineStyles';

const styles = () => ({
  italic: {
    fontStyle: 'italic',
  },
});

type Props = {
  classes: Object,
  tableHead: Array<Object>,
  tableData: Array<Object>,
  tableHeaderColor: string,
  PortfolioStore: Object,
};

type State = {
  order: string,
  orderBy: string,
};

function getSorting(order: string, orderBy: string) {
  return order === 'desc'
    ? (a: Object, b: Object) => (b[orderBy] < a[orderBy] ? -1 : 1)
    : (a: Object, b: Object) => (a[orderBy] < b[orderBy] ? -1 : 1);
}


const findInvestor = (portfolioStore: Object, transaction: Object) => {
  if (transaction.investorId !== null) {
    if (portfolioStore.currentPortfolioInvestors.length > 0) {
      return portfolioStore.currentPortfolioInvestors.find((item: Object) => item.id === transaction.investorId).name;
    }
    return 'unassigned';
  }
  return 'unassigned';
};

function getInvestorSortableObject(portfolioStore: Object) {
  return (transaction: Object) => Object.assign({}, {
    id: transaction.id,
    name: findInvestor(portfolioStore, transaction),
    transactionDate: new Date(transaction.txTimestamp).getTime(),
    type: transaction.type === 'd' ? 'DEPOSIT' : 'WITHDRAW',
    amountUSD: transaction.totalValueUSD.toFixed(2),
    sharePrice: transaction.currentSharePriceUSD.toFixed(2),
    shares: transaction.sharesCreated !== null ? transaction.sharesCreated.toFixed(2) : transaction.sharesLiquidated.toFixed(2),
    // comission: investor.amountInUSD < 0 ?
    //   Math.abs((investor.amountInUSD * portfolioStore.currentPortfolioInvestors[findInvestorID(investor)].managementFee) / 100) : 0,
    transaction,
  });
}

const TableHeader = ({
  onRequestSort,
  order,
  orderBy,
  classes,
  tableHead,
  tableHeaderColor,
}: {
  onRequestSort: Function,
  order: string,
  orderBy: string,
  classes: Object,
  tableHead: Array<Object>,
  tableHeaderColor: string,
}) => {
  const createSortHandler = (property: string) => (event: Object) => {
    onRequestSort(event, property);
  };
  return (
    <TableHead className={classes[`${tableHeaderColor}TableHeader`]}>
      <TableRow>
        {tableHead.map((headerItem: Object) => (
          <TableCell
            className={`${classes.tableCell} ${classes.tableHeadCell}`}
            key={headerItem.id}
            numeric={headerItem.numeric}
            padding={headerItem.disablePadding ? 'none' : 'default'}
            sortDirection={orderBy === headerItem.id ? order : false}
          >
            <Tooltip
              title="Sort"
              placement={headerItem.numeric ? 'bottom-end' : 'bottom-start'}
              enterDelay={300}
            >
              <TableSortLabel
                active={orderBy === headerItem.id}
                direction={order}
                onClick={createSortHandler(headerItem.id)}
              >
                {headerItem.label}
              </TableSortLabel>
            </Tooltip>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
};


@inject('PortfolioStore')
@observer
class AllInvestorTable extends React.Component<Props, State> {
  state = {
    order: 'desc',
    orderBy: 'dateOfEntry',
  };

  handleRequestSort = (event: Object, property: string) => {
    const orderBy = property;
    let order = 'desc';

    if (this.state.orderBy === property && this.state.order === 'desc') {
      order = 'asc';
    }

    this.setState({ order, orderBy });
  };

  render() {
    const { classes, tableHead, tableData, tableHeaderColor, PortfolioStore } = this.props;
    const { order, orderBy } = this.state;


    const header = (
      <TableHeader
        order={order}
        orderBy={orderBy}
        onRequestSort={this.handleRequestSort}
        tableHead={tableHead}
        classes={classes}
        tableHeaderColor={tableHeaderColor}
      />
    );

    return (
      <div className={classes.tableResponsive}>
        <Table className={classes.table}>
          {tableHead !== undefined ? header : null}
          <TableBody>
            {tableData
              .map(getInvestorSortableObject(PortfolioStore))
              .sort(getSorting(order, orderBy))
              .map((prop: Object) => (
                <TableRow key={uuid()}>
                  {Object.keys(prop).map((el: Object, key: number) => {
                    if (key > 7) return null;
                    if (key === 1) {
                      return (
                        <TableCell
                          className={`${classes.tableCell} ${prop.type === 'DEPOSIT' ? classes.positive : classes.negative}
                        ${prop.name === 'unassigned' ? classes.italic : ''}`}
                          key={uuid()}
                        >
                          {prop[el]}
                        </TableCell>
                      );
                    } if (key === 2) {
                      return (
                        <TableCell className={`${classes.tableCell} ${prop.type === 'DEPOSIT' ? classes.positive : classes.negative}`} key={uuid()}>
                          {moment(prop[el]).format('MMMM DD, YYYY HH:mm')}
                        </TableCell>
                      );
                    } if (key === 7) {
                      return (
                        <TableCell className={`${classes.tableCell}`} key={uuid()}>
                          {
                            prop.name === 'unassigned' ?
                              <AssignInvestor transaction={prop.transaction} />
                              :
                              null
                          }
                        </TableCell>
                      );
                    } else {
                      return (
                        <TableCell className={`${classes.tableCell} ${prop.type === 'DEPOSIT' ? classes.positive : classes.negative}`} key={uuid()}>
                          {prop[el]}
                        </TableCell>
                      );
                    }
                  })}
                </TableRow>
              ))}
          </TableBody>
        </Table>
      </div>
    );
  }
}

const combinedStyles = combineStyles(tableStyle, styles);

export default withStyles(combinedStyles)(AllInvestorTable);
