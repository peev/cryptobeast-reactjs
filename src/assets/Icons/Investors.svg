<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32.43 27.93"><defs><style>.cls-1,.cls-2{fill:none;stroke-miterlimit:10;}.cls-1{stroke:#fff;}.cls-2{stroke:#0bbcd5;}</style></defs><title>Asset 2</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1"><polygon class="cls-1" points="31.47 14.27 23.51 19.62 19.66 22.2 19.66 22.21 12.03 27.33 0.93 20.21 3.72 18.44 12.03 23.77 19.66 18.64 19.9 18.49 19.9 21.79 23.28 19.5 23.32 16.19 23.53 16.05 28.59 12.65 31.47 14.27"/><polygon class="cls-1" points="23.55 12.2 23.53 16.05 23.32 16.19 23.36 12.6 19.9 14.91 19.9 18.49 19.66 18.64 12.03 23.77 3.72 18.44 0.93 16.66 3.94 14.75 6.73 16.54 12.03 19.93 19.74 14.75 23.55 12.2"/><polygon class="cls-1" points="31.47 10.72 28.59 12.65 23.53 16.05 23.55 12.2 25.48 10.9 28.36 8.97 31.47 10.72"/><polygon class="cls-1" points="19.74 14.75 12.03 19.93 6.73 16.54 3.94 14.75 0.93 12.82 8.57 7.97 8.58 7.97 11.67 9.85 14.54 11.59 19.74 14.75"/><path class="cls-2" d="M25.48,10.9"/><polyline class="cls-1" points="25.48 10.9 23.55 12.2 19.74 14.75 14.54 11.59 11.67 9.85 8.58 7.97 12.52 5.48 15.61 7.36 18.47 9.09"/><path class="cls-2" d="M20.25,8"/><polyline class="cls-1" points="18.47 9.09 15.61 7.36 12.52 5.48 20.25 0.58 31.47 6.88 28.36 8.97 25.48 10.9"/><line class="cls-1" x1="23.51" y1="12.15" x2="18.48" y2="9.09"/></g></g></svg>