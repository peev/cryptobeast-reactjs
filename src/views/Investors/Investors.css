    .InvPaper {
        max-height: 500px;
        width: 100%;
        margin-top: 20px;
        padding: 20px;
    }

    .InvPaperTable {
        max-height: 500px;
        width: 100%;
        margin-top: 20px;
    }

    .InvPaper button {
        float: right;
    }

    .InvButtonsGroup {
        width: 100%;
        display: inline-flex;
        justify-content: space-between;
        margin-top: 20px;
    }

    .InvButtonsGroup button {
        width: 180px;
        color: #fff;
        background-color: #5e6779;
    }

    .InvButtonsGroup button:hover {
        width: 180px;
        color: #fff;
        background-color: #666666;
    }

    .InvCardsGroup {
        width: 100%;
        display: inline-flex;
        justify-content: space-between;
        margin-top: 20px;
    }