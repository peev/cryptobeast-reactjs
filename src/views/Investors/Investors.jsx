import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'material-ui';
import Paper from 'material-ui/Paper';

import RegularButton from '../../components/CustomButtons/Button';
import AddInvestorWrapped from '../../components/Modal/InvestorModals/AddInvestor';
import InvestorDepositWrapped from '../../components/Modal/InvestorModals/InvestorDeposit';
import SelectInvestor from '../../components/Selectors/SelectInvestor';
import GenericTable from '../../components/GenericTable/GenericTable';
import EditInvestorWrapped from '../../components/Modal/InvestorModals/EditInvestor';
import InvestorWithdrawWrapped from '../../components/Modal/InvestorModals/InvestorWithdraw';
// test db
// const { ipcRenderer } = window.require('electron');
import './Investors.css';
import TotalInvestorsWrapped from '../../components/Modal/InvestorModals/TotalInvestors';
import SharesInCirculationWrapped from '../../components/Modal/InvestorModals/SharesInCirculation';
import CurrentSharePriceWrapped from '../../components/Modal/InvestorModals/CurrentSharePrice';
import TotalUSDEquivWrapped from '../../components/Modal/InvestorModals/TotalUSDEquiv';

const dropStyle = {
  width: '100%'
};

class Investors extends React.Component {
  state = {
    direction: 'row'
    // justify: 'flex-end',
    // alignItems: 'center',
  };

  render() {
    const { direction } = this.state;
    // const { alignItems, direction, justify } = this.state;

    return (
      <div>
        <Grid container className="InvCardsGroup">
          <TotalInvestorsWrapped />
          <SharesInCirculationWrapped />
          <CurrentSharePriceWrapped />
          <TotalUSDEquivWrapped />
        </Grid>
        <Grid container className="InvButtonsGroup">
          <AddInvestorWrapped />
          <InvestorDepositWrapped />
          <InvestorWithdrawWrapped />

          <EditInvestorWrapped />
        </Grid>
        <Grid container className="InvGrid">
          <Paper container className="InvPaper">
            <Grid direction={direction}>
              <h5>INDIVIDUAL SUMMARY</h5>
            </Grid>

            <Grid container>
              <Grid item xs={3}>
                <SelectInvestor style={dropStyle} />
              </Grid>
              <Grid item xs={3}>
                <p>Shares Held:</p>
                <p>Weighted entry price:</p>
                <p>Current Share Price:</p>
              </Grid>
              <Grid item xs={3}>
                <p>Shares Held:</p>
                <p>Weighted entry price:</p>
                <p>Current Share Price:</p>
              </Grid>
              <Grid item xs={3}>
                <p>Shares Held:</p>
                <p>Weighted entry price:</p>
                <p>Current Share Price:</p>
              </Grid>
            </Grid>
            <RegularButton color="primary">Export</RegularButton>
          </Paper>
        </Grid>
        <Grid container className="InvGrid">
          <Paper className="InvPaperTable">
            <GenericTable
              tableHead={[
                'ID',
                'Name',
                'Date of Entry',
                'Transaction date',
                'Amount (USD)',
                'Share price',
                'New/Liquidated Shares'
              ]}
              tableData={[
                [
                  '1',
                  'SomeINvestor',
                  'A day',
                  'Transaction Dates',
                  '1$',
                  '1$',
                  'Test'
                ]
              ]}
            />
          </Paper>
        </Grid>
      </div>
    );
  }
}

export default Investors;
