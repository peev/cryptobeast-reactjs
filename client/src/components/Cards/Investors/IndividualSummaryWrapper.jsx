// @flow
import React, { Component } from 'react';
import { withStyles } from 'material-ui';
import { inject, observer } from 'mobx-react';

import IndividualSummary from './IndividualSummary';
import { RegularCard } from './../../../components';
import AllInvestorTable from './../../CustomTables/Investors/AllInvestorTable';
// import ExportPdfButton from './../../CustomButtons/ExportPdfButton';

const styles = () => ({
  tablePosition: {
    marginTop: '80px',
    '& th': {
      fontWeight: '700',
    },
  },
});

type Props = {
  classes: Object,
  InvestorStore: Object,
  PortfolioStore: Object,
};

@inject('InvestorStore', 'PortfolioStore')
@observer
class IndividualSummaryWrapper extends Component<Props> {
  render() {
    const { classes, InvestorStore, PortfolioStore } = this.props;

    const data = InvestorStore.selectedInvestorIndividualSummary
      ? PortfolioStore.currentPortfolioTransactions
        .filter((t: Object) => t.investorId === InvestorStore.selectedInvestorIndividualSummary.id)
      : PortfolioStore.currentPortfolioTransactions;

    const investorsTable = (
      <AllInvestorTable
        tableData={data}
        tableHead={[
          // 'ID',
          'Name',
          'Date of Entry',
          'Transaction date',
          'Amount (USD)',
          'Share price',
          'Shares',
          'Commission',
        ]}
      />
    );

    return (
      <RegularCard
        cardTitle="INDIVIDUAL SUMMARY"
        // button={<ExportPdfButton />}
        content={
          <div>
            <IndividualSummary />
            <div className={classes.tablePosition}>
              {investorsTable}
            </div>
          </div>
        }
      />
    );
  }
}

export default withStyles(styles)(IndividualSummaryWrapper);
