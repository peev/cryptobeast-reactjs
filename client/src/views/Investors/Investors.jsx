// @flow
import React from 'react';
import { withStyles, Grid } from 'material-ui';

import TotalInvestorsWrapped from '../../components/Modal/InvestorModals/TotalInvestors';
import SharesInCirculationWrapped from '../../components/Modal/InvestorModals/SharesInCirculation';
import CurrentSharePrice from '../../components/Modal/InvestorModals/CurrentSharePrice';
import TotalFeePotential from '../../components/Modal/InvestorModals/TotalFeePotential';

import AddInvestorWrapped from '../../components/Modal/InvestorModals/AddInvestor';
import InvestorDepositWrapped from '../../components/Modal/InvestorModals/InvestorDeposit';
import InvestorWithdrawWrapped from '../../components/Modal/InvestorModals/InvestorWithdraw';
import EditInvestorWrapped from '../../components/Modal/InvestorModals/EditInvestor';

import IndividualSummaryWrapper from '../../components/Cards/Investors/IndividualSummaryWrapper';


const styles = () => ({
  herderTopButton: {
    padding: '25px 50px',
    '& button': {
      display: 'flex',
      width: '100%',
      margin: '0',
    },
  },
  herderBottomButton: {
    padding: '25px 50px',
    '& button': {
      display: 'flex',
      width: '100%',
      color: '#fff',
      backgroundColor: '#5e6779',
      margin: '0',
      '&:hover': {
        backgroundColor: '#666666',
      },
    },
  },
  upperPart: {
    justifyContent: 'center',
    margin: '-25px 0 25px 0',
  },
  upperPartSingleRow: {
    justifyContent: 'center',
  },
});


const Investors = ({ classes }: Object) => (
  <Grid container>
    <Grid container className={classes.upperPart}>
      <Grid container className={classes.upperPartSingleRow}>
        <Grid item xs={12} sm={9} md={5} lg={3} className={classes.herderTopButton}>
          <TotalInvestorsWrapped />
        </Grid>
        <Grid item xs={12} sm={9} md={5} lg={3} className={classes.herderTopButton}>
          <SharesInCirculationWrapped />
        </Grid>
        <Grid item xs={12} sm={9} md={5} lg={3} className={classes.herderTopButton}>
          <CurrentSharePrice />
        </Grid>
        <Grid item xs={12} sm={9} md={5} lg={3} className={classes.herderTopButton}>
          <TotalFeePotential />
        </Grid>
      </Grid>

      <Grid container className={classes.upperPartSingleRow}>
        <Grid item xs={12} sm={9} md={5} lg={3} className={classes.herderBottomButton}>
          <AddInvestorWrapped />
        </Grid>
        <Grid item xs={12} sm={9} md={5} lg={3} className={classes.herderBottomButton}>
          <InvestorDepositWrapped />
        </Grid>
        <Grid item xs={12} sm={9} md={5} lg={3} className={classes.herderBottomButton}>
          <InvestorWithdrawWrapped />
        </Grid>
        <Grid item xs={12} sm={9} md={5} lg={3} className={classes.herderBottomButton}>
          <EditInvestorWrapped />
        </Grid>
      </Grid>
    </Grid>

    <Grid container>
      <Grid item xs={12} sm={12} md={12}>
        <IndividualSummaryWrapper />
      </Grid>
    </Grid>
  </Grid>
);

export default withStyles(styles)(Investors);
