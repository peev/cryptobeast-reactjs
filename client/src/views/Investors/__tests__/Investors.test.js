import React from 'react';
import { shallow } from 'enzyme';
import Investors from './../Investors';

describe('Views/Investors.jsx', () => {
  const investors = shallow(<Investors />);

  it('should render without crashing', () => {
    expect(investors).toMatchSnapshot();
  });
});
