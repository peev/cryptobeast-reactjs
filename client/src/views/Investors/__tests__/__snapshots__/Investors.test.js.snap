// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views/Investors.jsx should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Investors) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "herderBottomButton": "Investors-herderBottomButton-02",
        "herderTopButton": "Investors-herderTopButton-01",
        "upperPart": "Investors-upperPart-03",
        "upperPartSingleRow": "Investors-upperPartSingleRow-04",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "herderBottomButton": "Investors-herderBottomButton-02",
          "herderTopButton": "Investors-herderTopButton-01",
          "upperPart": "Investors-upperPart-03",
          "upperPartSingleRow": "Investors-upperPartSingleRow-04",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
